_source/_posts/2020-10-22-set-up-a-mute-indicator-light-for-zoom-with-hammerspoon.md

```diff
@@ -0,0 +1,283 @@
+
+```shell
+mkdir ~/blink
+```
+
+Next, `cd` to the directory you just created and make a new virtualenv for Python.
+
+```shell
+cd ~/blink
+virtualenv venv
+```
+
+At this point, the virtualenv should be activated. Your prompt will contain `(venv)` if so. If you don't see `(venv)` in your prompt, try running:
+
+```shell
+source venv/bin/activate
+```
+
+Now you're ready to install the Python SDK for the blink(1) light:
+
+```shell
+pip install blink1
+```
+
+With the blink1 package installed, we're ready to write some code that uses it. Below is the source code for a small Python program that I wrote to control the blink(1) light from the command line:
+
+```python
+#!/usr/bin/env python3
+from blink1.blink1 import Blink1
+import sys
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-r", "--red", default=0, type=int, help="Value for red (0 - 255)")
+parser.add_argument("-g", "--green", default=0, type=int, help="Value for green (0 - 255)")
+parser.add_argument("-b", "--blue", default=0, type=int, help="Value for blue (0 - 255)")
+parser.add_argument("-d", "--delay", default=50, type=int, help="Light shift time in milliseconds")
+args = parser.parse_args()
+
+b1 = Blink1()
+b1.fade_to_rgb(args.delay, args.red, args.green, args.blue)
+```
+
+Copy the code above into a file named `blink.py` and then try running it from the command line:
+
+```shell
+python blink.py --green 128
+```
+
+This should turn the light green at about half brightness.
+
+Turn it off with this command:
+
+```shell
+python blink.py
+```
+
+
+## Connecting Hammerspoon to the blink(1) USB LED
+
+Now that you have a working Hammerspoon setup and a working Python script to control the blink(1) light, it's time to tie them together!
+
+Start by selecting **Open Config** from the Hammerspoon menu icon, then edit the `~/.hammerspoon/init.lua` file to look like the code below. Make sure to replace the path to the Python scripts with the path for your system!
+
+```lua
+updateZoomStatus = function(event)
+  hs.printf("updateZoomStatus(%s)", event)
+  if (event == "from-running-to-meeting") then
+    zoomStatusMenuBarItem:returnToMenuBar()
+  elseif (event == "muted") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py" "--red" "128"]])
+    zoomStatusMenuBarItem:setTitle("🔴")
+  elseif (event == "unmuted") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py" "--green" "128"]])
+    zoomStatusMenuBarItem:setTitle("🟢")
+  elseif (event == "from-meeting-to-running") then
+    hs.execute([["/Users/joel/Projects/mute/venv/bin/python" "/Users/joel/Projects/mute/blink.py"]])
+    zoomStatusMenuBarItem:removeFromMenuBar()
+  end
+end
+```
+
+Once you've finished modifying the `updateZoomStatus` function, save the file, then use the Hammerspoon menu icon to select the "Reload Config" menu item.
```

Comment:
Change to `updateZoomStatus()` (since it's a method or function) and bold Reload Config (since it's something you click on.